packages/flutter/test/rendering/wrap_test.dart

```diff
@@ -12,17 +12,146 @@ void main() {
     expect(
       renderWrap.toStringDeep(minLevel: DiagnosticLevel.info),
       equalsIgnoringHashCodes(
-        'RenderWrap#00000 NEEDS-LAYOUT NEEDS-PAINT DETACHED\n'
-        '   parentData: MISSING\n'
-        '   constraints: MISSING\n'
-        '   size: MISSING\n'
-        '   direction: horizontal\n'
-        '   alignment: start\n'
-        '   spacing: 0.0\n'
-        '   runAlignment: start\n'
-        '   runSpacing: 0.0\n'
-        '   crossAxisAlignment: 0.0\n'
-      ),
+          'RenderWrap#00000 NEEDS-LAYOUT NEEDS-PAINT DETACHED\n'
+          '   parentData: MISSING\n'
+          '   constraints: MISSING\n'
+          '   size: MISSING\n'
+          '   direction: horizontal\n'
+          '   alignment: start\n'
+          '   spacing: 0.0\n'
+          '   runAlignment: start\n'
+          '   runSpacing: 0.0\n'
+          '   crossAxisAlignment: 0.0\n'),
     );
   });
+
+  test(
+    'Compute intrinsic height test',
+    () {
+      final List<RenderBox> children = <RenderBox>[
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+      ];
+
+      _intrinsicHeightDefaultTest(children, 245, 165);
+      _intrinsicHeightDefaultTest(children, 250, 80);
+      _intrinsicHeightDefaultTest(children, 80, 250);
+      _intrinsicHeightDefaultTest(children, 79, 250);
+      _intrinsicHeightDefaultTest(children, 245, 165, isMax: false);
+      _intrinsicHeightDefaultTest(children, 250, 80, isMax: false);
+      _intrinsicHeightDefaultTest(children, 80, 250, isMax: false);
+      _intrinsicHeightDefaultTest(children, 79, 250, isMax: false);
+    },
+  );
+
+  test(
+    'Compute intrinsic width test',
+    () {
+      final List<RenderBox> children = <RenderBox>[
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+        RenderConstrainedBox(
+          additionalConstraints:
+              const BoxConstraints(minWidth: 80, minHeight: 80),
+        ),
+      ];
+
+      _intrinsicWidthDefaultTest(children, 245, 165);
```

Comment:
use named parameters for the 245 and 165 to make clear what those numbers mean.

Reply:
Even better: To make the test easier to understand remove the indirection via _intrinsicWidthDefaultTest.

The lines here should just be
```dart
expect(renderWrap.computeMaxIntrinsicWidth(245), 165);
```

You can do all the setup you do in _intrinsicWidthDefaultTest and _prepareWrapForComputeIntrinsicSize directly in this test.

Reply:
@goderbauer 
Thank you. I was try to make tests more readable and structurable, but really they became very hard understand. I refactor this and add some tests to cover cases when only one run.